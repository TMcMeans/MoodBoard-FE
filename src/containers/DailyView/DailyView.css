/*COMPONENT STYLES*/
.daily-view {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100vh;
  margin: 0;
}

.analytical-bg {
  background: rgba(129,220,176, 0.62);
}

.joy-bg {
  background: rgba(227,186,86, 0.8);
}

.fear-bg {
  background: rgba(237,158,90, 0.8);
}

.confident-bg {
  background: rgba(98,74,121, 0.8);
}

.tentative-bg {
  background: rgba(235,123,162, 0.8);
}

.anger-bg {
  background: rgba(221,134,134, 0.9);
}

.sadness-bg {
  background: rgba(106,164,203, 0.9);
}

/*DATE*/
.daily-date {
  color: #ffffff;
  margin: 0;
}

/*DAILY TONE*/
.tone-word {
  font-weight: bold;
  font-size: 54px;
  margin: 0 0 1.5rem;
}

.analytical {
  color: #4BCE8F;
  text-shadow: 0px 4px 4px rgba(84, 84, 84, 0.8);
}

.joy {
  color: #e0b344;
  text-shadow: 0px 4px 4px rgba(42, 42, 42, 0.8);
}

.fear {
  color: #e98632;
  text-shadow: 0px 4px 4px rgba(42, 42, 42, 0.8);
}

.confident {
  color: #3b1e58;
  text-shadow: 0px 2px 2px rgba(255, 255, 255, 0.3);
}

.tentative {
  color: #E02F6D;
  text-shadow: 0px 4px 4px rgba(42, 42, 42, 0.8);
}

.anger {
  color: #C73636;
  text-shadow: 0px 2px 2px rgba(255, 255, 255, 0.4);
}

.sadness {
  color: #2b7db5;
  text-shadow: 0px 4px 4px rgba(42, 42, 42, 0.8);
}

/*JOURNAL AND AFFIRMATION*/
.daily-entry,
.daily-affirmation {
  background-color: #f5f1e7;
  background-image: url('../../assets/paper2.png');
  border: 1px solid #b5b5b5;
  border-radius: 5px;
  color: #282625;
  font-size: 1.4rem;
  margin: 0 0 1rem;
  padding: 20px;
  overflow: scroll;
  text-align: left;
  width: 75%;
}

.daily-entry {
  height: 30%;
}

/*AFFIRMATION*/
.daily-affirmation-text {
  margin: 0;
}

/*BUTTONS*/
.btn-wrapper {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 1rem;
  width: 65%;
}

.left-arrow,
.back-to-calendar-btn,
.right-arrow {
  font-size: 1.3rem;
  line-height: 1rem;
  height: 3rem;
}
