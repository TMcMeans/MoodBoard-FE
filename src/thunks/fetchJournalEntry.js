import { addJournalEntry, hasErrored } from '../actions';

export const fetchJournalEntry = url => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const journal_entry = await response.json();
      dispatch(getJournalEntry(journal_entry));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
