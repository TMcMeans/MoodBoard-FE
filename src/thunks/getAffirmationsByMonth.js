import { hasErrored } from '../actions';

export const getAffirmationsByMonth = url => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const affirmations = await response.json();
      return affirmations;
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
