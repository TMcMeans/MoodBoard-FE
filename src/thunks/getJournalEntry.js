import { saveJournalEntry, hasErrored } from '../actions';

export const getJournalEntry = url => {
      console.log('running')
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const journal_entry = await response.json();
      console.log(journal_entry)
      dispatch(saveJournalEntry(journal_entry));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
