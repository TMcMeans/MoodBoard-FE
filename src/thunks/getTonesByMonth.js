import { saveMonthTones, hasErrored } from '../actions';

export const getTonesByMonth = (url) => {
  return async dispatch => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const tone_data = await response.json();
      await dispatch(saveMonthTones(tone_data.data));
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
