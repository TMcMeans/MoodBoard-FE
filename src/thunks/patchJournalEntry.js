import { hasErrored } from '../actions';

export const patchJournalEntry = (url, entry_text) => {
  return async dispatch => {
    try {
      const response = await fetch(url, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          journal_entry: { entry_text }
        })
      });
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
