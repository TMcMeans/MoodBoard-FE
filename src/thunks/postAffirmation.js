import { hasErrored } from '../actions';

export const postAffirmation = (url, affirmation_text) => {
  return async dispatch => {
    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          affirmation: { affirmation_text }
        })
      });
      if (!response.ok) {
        throw Error(response.statusText);
      }
    } catch (error) {
      dispatch(hasErrored(error.message));
    }
  };
};
