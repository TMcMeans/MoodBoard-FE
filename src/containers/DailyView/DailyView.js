import React, { Component } from 'react';
import Logo from '../../components/Logo/Logo';
import Button from '../../components/Button/Button';

import './DailyView.css';

class DailyView extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    const toneWord = 'sadness';
    const entry =
      'Now, when you do this without getting punched in the chest, you will have more fun. Get me a vodka rocks. And a piece of toast. Now, when you do this without getting punched in the chest, you will have more fun. Get me a vodka rocks...';

    const affirmation = 'I got this.';
    const dateText = 'June 16, 2018';
    return (
      <div className="daily-view sadness-bg">
        <Logo />
        <h2 className="daily-date">{dateText}</h2>
        <h1 className="tone-word sadness">{toneWord}</h1>
        <div class="daily-entry">{entry}</div>
        <div class="daily-affirmation">
          <p className="daily-affirmation-text">{affirmation}</p>
        </div>
        <div className="btn-wrapper">
          <Button text="<" />
          <Button text="back" />
          <Button text=">" />
        </div>
      </div>
    );
  }
}

export default DailyView;
