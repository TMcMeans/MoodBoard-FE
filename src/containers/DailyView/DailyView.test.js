import React from 'react';
import { shallow } from 'enzyme';
import { DailyView, mapStateToProps, mapDispatchToProps } from './DailyView';
import { getJournalEntryByDate } from '../../thunks/getJournalEntryByDate';

jest.mock('../../thunks/getJournalEntryByDate');

describe('DailyView', () => {
  let wrapper;
  let mockJournal = {
    journal_entry_text: 'I am angry',
    primary_tone: 'angry'
  };
  beforeEach(() => {
    wrapper = shallow(<DailyView journal={mockJournal} />);
  });

  it('should render a snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should return a journal object', () => {
    const mockState = { journal: { entry_text: 'I am so sad.' } };
    const expected = { entry_text: 'I am so sad.' };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps.journal).toEqual(expected);
  });

  it('mapDispatchToProps should dispatch getJournalEntryByDate', () => {
    const mockDispatch = jest.fn();
    getJournalEntryByDate.mockImplementation(() => {});

    const expected = getJournalEntryByDate('www.journals.com');

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getJournalEntryByDate('www.journals.com');

    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
