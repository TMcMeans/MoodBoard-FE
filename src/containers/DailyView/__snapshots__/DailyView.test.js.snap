// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyView should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DailyView
    journal={
      Object {
        "journal_entry_text": "I am angry",
        "primary_tone": "angry",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h2
          className="daily-date"
        >
          February 17, 2019
        </h2>,
        <h1>
          angry
        </h1>,
        <div
          className="daily-entry"
        >
          I am angry
        </div>,
        <div
          className="daily-affirmation"
        >
          <p
            className="daily-affirmation-text"
          />
        </div>,
        <div
          className="btn-wrapper"
        >
          <button
            className="standard-btn dark left-arrow"
          >
            &lt;
          </button>
          <Link
            replace={false}
            to="/calendarview"
          >
            <button
              className="standard-btn dark back-to-calendar-btn"
            >
              calendar
            </button>
          </Link>
          <button
            className="standard-btn dark right-arrow"
          >
            &gt;
          </button>
        </div>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "February 17, 2019",
          "className": "daily-date",
        },
        "ref": null,
        "rendered": "February 17, 2019",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "angry",
          "className": undefined,
        },
        "ref": null,
        "rendered": "angry",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am angry",
          "className": "daily-entry",
        },
        "ref": null,
        "rendered": "I am angry",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="daily-affirmation-text"
          />,
          "className": "daily-affirmation",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "daily-affirmation-text",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="standard-btn dark left-arrow"
            >
              &lt;
            </button>,
            <Link
              replace={false}
              to="/calendarview"
            >
              <button
                className="standard-btn dark back-to-calendar-btn"
              >
                calendar
              </button>
            </Link>,
            <button
              className="standard-btn dark right-arrow"
            >
              &gt;
            </button>,
          ],
          "className": "btn-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "<",
              "className": "standard-btn dark left-arrow",
            },
            "ref": null,
            "rendered": "<",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="standard-btn dark back-to-calendar-btn"
              >
                calendar
              </button>,
              "replace": false,
              "to": "/calendarview",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "calendar",
                "className": "standard-btn dark back-to-calendar-btn",
              },
              "ref": null,
              "rendered": "calendar",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": ">",
              "className": "standard-btn dark right-arrow",
            },
            "ref": null,
            "rendered": ">",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h2
            className="daily-date"
          >
            February 17, 2019
          </h2>,
          <h1>
            angry
          </h1>,
          <div
            className="daily-entry"
          >
            I am angry
          </div>,
          <div
            className="daily-affirmation"
          >
            <p
              className="daily-affirmation-text"
            />
          </div>,
          <div
            className="btn-wrapper"
          >
            <button
              className="standard-btn dark left-arrow"
            >
              &lt;
            </button>
            <Link
              replace={false}
              to="/calendarview"
            >
              <button
                className="standard-btn dark back-to-calendar-btn"
              >
                calendar
              </button>
            </Link>
            <button
              className="standard-btn dark right-arrow"
            >
              &gt;
            </button>
          </div>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "February 17, 2019",
            "className": "daily-date",
          },
          "ref": null,
          "rendered": "February 17, 2019",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "angry",
            "className": undefined,
          },
          "ref": null,
          "rendered": "angry",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I am angry",
            "className": "daily-entry",
          },
          "ref": null,
          "rendered": "I am angry",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="daily-affirmation-text"
            />,
            "className": "daily-affirmation",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "daily-affirmation-text",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="standard-btn dark left-arrow"
              >
                &lt;
              </button>,
              <Link
                replace={false}
                to="/calendarview"
              >
                <button
                  className="standard-btn dark back-to-calendar-btn"
                >
                  calendar
                </button>
              </Link>,
              <button
                className="standard-btn dark right-arrow"
              >
                &gt;
              </button>,
            ],
            "className": "btn-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "<",
                "className": "standard-btn dark left-arrow",
              },
              "ref": null,
              "rendered": "<",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="standard-btn dark back-to-calendar-btn"
                >
                  calendar
                </button>,
                "replace": false,
                "to": "/calendarview",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "calendar",
                  "className": "standard-btn dark back-to-calendar-btn",
                },
                "ref": null,
                "rendered": "calendar",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": ">",
                "className": "standard-btn dark right-arrow",
              },
              "ref": null,
              "rendered": ">",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
